### ISSUE TITLE
Get-Wmiobject -ComputerName  access denied

### ISSUE THREAD

**Issue by santosshen** [jborean93/pypsrp#30]
from pypsrp.powershell import PowerShell, RunspacePool
from pypsrp.wsman import WSMan

wsman = WSMan(server="serverA", username=r"XXX",password="XXX", ssl=False)
with RunspacePool(wsman) as pool:
    ps = PowerShell(pool)

    ps.add_script("Get-Wmiobject -ComputerName serverB -Class win32_bios")
    output = ps.invoke()

print("HAD ERRORS: %s" % ps.had_errors)
print("OUTPUT:\n%s" % "\n".join([str(s) for s in output]))
print("ERROR:\n%s" % "\n".join([str(s) for s in ps.streams.error]))
print("DEBUG:\n%s" % "\n".join([str(s) for s in ps.streams.debug]))
print("VERBOSE:\n%s" % "\n".join([str(s) for s in ps.streams.verbose]))

‘’‘
HAD ERRORS: True
OUTPUT:

ERROR:
access denied。 (Exception from HRESULT: 0x80070005 (E_ACCESSDENIED))
DEBUG:

VERBOSE:
···
I want  serverA  Remote execution to serverB

**Comment by santosshen** [comment]
[图片]

**Comment by santosshen** [comment]
![image](https://user-images.githubusercontent.com/25058112/53871997-3d29b300-4038-11e9-8707-790f4d4108b9.png)

**Comment by jborean93** [comment]
By default your credentials are not sent to the WinRM process so it cannot then connect to another server. It will fail with an access is denied because it is authenticating with an Anonymous user. This is explained more at https://blogs.technet.microsoft.com/ashleymcglone/2016/08/30/powershell-remoting-kerberos-double-hop-solved-securely/.

The easiest way to solve this is to use `auth='credssp'` when creating the `WSMan` object, if you are working with kerberos you could do `auth='kerberos, negotiate_delegate=True` to ensure you are using Kerberos delegation. The other option is to use explicit credentials in your script when connecting to that host like;

```
from pypsrp.complex_objects import PSCredential
from pypsrp.powershell import PowerShell, RunspacePool
from pypsrp.wsman import WSMan

username = r'XXX'
password = r'XXX'

wsman = WSMan(server="serverA", username=username, password=password, ssl=False)
with RunspacePool(wsman) as pool:
    # Required so that we can securely exchange the pass in the PSCredential without it
    # being sent in plaintext
    pool.exchange_keys()

    # You don't need to use these creds, it could be something else
    credential = PSCredential(username=username, password=password)

    ps = PowerShell(pool)

    # Set the variable $cred to equal our credential so it can be used in the script. The
    # object is serialized as a SecureString so we are not exposing the password in
    # plaintext
    ps.add_cmdlet("Set-Variable").add_parameters({"Name": "cred", "Value": credential})

    # We then use the Credential defined above in our script when creating the CimSession
    ps.add_statement().add_script('''
$session = New-CimSession -ComputerName serverB -Credential $cred
Get-CimInstance -ClassName Win32_Bios -CimSession $session''')

    output = ps.invoke()

print("HAD ERRORS: %s" % ps.had_errors)
print("OUTPUT:\n%s" % "\n".join([str(s) for s in output]))
print("ERROR:\n%s" % "\n".join([str(s) for s in ps.streams.error]))
print("DEBUG:\n%s" % "\n".join([str(s) for s in ps.streams.debug]))
print("VERBOSE:\n%s" % "\n".join([str(s) for s in ps.streams.verbose]))
```

**Comment by santosshen** [comment]
Hi jborean93,
     Can auth='credssp'  and auth='kerberos, negotiate_delegate=True   provide examples
     What needs to be configured on the serverA serverB
 @jborean93 
     
![image](https://user-images.githubusercontent.com/25058112/54002514-cbfa1500-4189-11e9-9bb3-19a7c38808b2.png)
or
[Error] Failed to connect to WMI service in computer  serverB

  ths

**Comment by jborean93** [comment]
CredSSP is the simplest, just set `auth='credssp'` and put your username and password to the user you want. You will have to enable CredSSP auth on the server you are connecting to on WinRM (serverA), see https://docs.microsoft.com/en-us/powershell/module/microsoft.wsman.management/enable-wsmancredssp?view=powershell-6 for more info.

Kerberos delegation is a bit more difficult to test out. I found a bug when using delegation on Windows that was recently fixed in the [v0.3.1](https://github.com/jborean93/pypsrp/releases/tag/v0.3.1) release. Please make sure you have updated to that version before trying it out. It is also reliant on some configuration on the active directory side, if it has been restricted you won't be able to get it working.

**Comment by santosshen** [comment]
CredSSP is the simplest, just set auth='credssp'
server configured  Enable-WSManCredSSP -Role "Server"

ths
